### DISCUSSION TITLE
Why *a and *b are divided by 110?

### DISCUSSION THREAD

**Discussion by ZenithFlux** [junyanz/pytorch-CycleGAN-and-pix2pix#1433] [Q&A]
I think in colorization_dataset.py, the ab tensor should be divided by 128, not 110, as scikit-image implementation of L\*a\*b has a range of -128 to 127 for \*a and \*b.

The range of ab is -110 to 110 in [this](http://ai.stanford.edu/~ruzon/software/rgblab.html) Matlab implementation.So I think this is the source of 110. But i dont think this is valid for scikit, is it?

**Comment by junyanz** [comment]
This is a great question. Originally, we followed the Torch implementation of pix2pix. See this [line](https://github.com/phillipi/pix2pix/blob/89ff2a81ce441fbe1f1b13eca463b87f1e539df8/data/donkey_folder.lua#L108). But I am not sure if the Lua image and skimage produce the same or different ab range. 
I did some Google search regarding the Lab range of skimage, but could not find a definite answer. This [post](https://stackoverflow.com/questions/25294141/cielab-color-range-for-scikit-image) mentioned -128, 128, but this [post](https://github.com/scikit-image/scikit-image/issues/1185) mentioned a narrow range, which is within -110, 110.
